build.gradle

```diff
@@ -865,15 +873,16 @@ def calculateVersion() {
   return version
 }
 
-def getCheckedOutGitCommitHash() {
+
+def getCheckedOutGitCommitHash(length = 8) {
   try {
     def gitFolder = "$projectDir/.git/"
     if (!file(gitFolder).isDirectory()) {
       // We are in a submodule.  The file's contents are `gitdir: <gitFolder>\n`.
       // Read the file, cut off the front, and trim the whitespace.
-      gitFolder = file(gitFolder).text.substring(8).trim() + "/"
+      gitFolder = file(gitFolder).text.substring(length).trim() + "/"
     }
-    def takeFromHash = 8
+    def takeFromHash = length
     /*
      * '.git/HEAD' contains either
      *      in case of detached head: the currently checked out commit hash
```

Comment:
suggestion: rename hash to sha for consistency?

Reply:
The only uses I see of sha are for shaTag and that is being used to refer to the combined string of gitSha-{hash}. So think it actually makes sense to stay as hash.